controller/sync.go

```diff
@@ -259,14 +259,19 @@ func (sc *syncContext) sync() {
 	// If no sync tasks were generated (e.g., in case all application manifests have been removed),
 	// the sync operation is successful.
 	if len(tasks) == 0 {
-		sc.setOperationPhase(v1alpha1.OperationSucceeded, "successfully synced")
+		sc.setOperationPhase(v1alpha1.OperationSucceeded, "successfully synced (no more tasks)")
 		return
 	}
 
 	// remove any tasks not in this wave
 	phase := tasks.phase()
 	wave := tasks.wave()
 
+	// if it is the last phase/wave and the only remaining tasks are non-hooks, the we are successful
+	// EVEN if those objects subsequently degraded
```

Comment:
This handles the common case where neither hooks or waves are used and a sync equates to simply an (asynchronous) `kubectl apply` of manifests, which succeeds immediately.